src/2dplane/ObstacleGrid.cpp

```diff
@@ -24,6 +28,38 @@ Vector2i ObstacleGrid::gridSquareForLocation(const Vector2f &loc) const {
                     loc.y() / height() * discretizedHeight());
 }
 
+float ObstacleGrid::nearestObstacleDist(const Vector2f &state, float maxDist) const {
+    //x and y are the indices of the cell that state is located in
+    float x = (state.x() / (_width / _discretizedWidth));
+    float y = (state.y() / (_height / _discretizedHeight));
+    int xSearchRad = maxDist * _discretizedWidth / (_width * SQRT2);
```

Comment:
I think your math is right for finding the x and y for a point diagonal from the center of your search, but consider how far out a point is that's directly to the right of your center point.  In this case, the difference in the `y` is zero, but the difference in `x` is `maxDist * _discretizedWidth / _width`.  The `SQRT2` would come into play if your search zone was a circle, but with the way you have it setup with your for loops, you're actually searching a rectangle.  A circular search might be more efficient, but I think the way you have it now is a little better because of its simplicity.  I don't think the `SQRT2` factor is needed though since you're not actually calculating a circle.